### DISCUSSION TITLE
[Help]: How to get the html node of the current slide?

### DISCUSSION THREAD

**Discussion by dehghani-mehdi** [davidjerleke/embla-carousel#1031] [Help] [answered]
### Summary

`emblaApi.selectedScrollSnap()` returns the index of the current slide, is there any way to get the HTML node of current slide?

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
@dehghani-mehdi like this:

```tsx
function getSelectedSlideNode(emblaApi) {
  const selectedIndex = emblaApi.selectedScrollSnap()
  return emblaApi.slideNodes()[selectedIndex]
}

// Usage on init:
const selectedSlideNode = getSelectedSlideNode(emblaApi)

// Usage when the selected slide changes:
emblaApi.on('select', (emblaApi) => {
  const selectedSlideNode = getSelectedSlideNode(emblaApi)
})
```

> [!IMPORTANT]  
> This only works if you don't group slides with `slidesToScroll`. If you do, let me know and I'll show how you get all slide nodes that belong to a specific index/group.
